### ISSUE TITLE
Groups of favorites should have persistent and portable visual appearance

### ISSUE THREAD

**Issue by Friptick** [osmandapp/OsmAnd#17178]
At present, OsmAnd allows the user to organize favorites (and track waypoints) into "groups" and to set a "default appearance" (i.e., color and icon) for each group. But, in the underlying GPX code, these attributes remain attached to each individual favorite and track waypoint, not to the group.

This cause problems:
 - Changing the appearance of a group is error-prone, because all the individual favorites or track waypoints need to be updated - and this will not affect any other tracks the user may have, resulting in unexpected behavior when the other tracks are opened
 - Changing the group of a single favorite (or track waypoint) is ambiguity-prone for the same reason
 - **It is not portable**: the user cannot use other software to read or write the visual appearance of track waypoints

In #10850 @vshcherb says that the current system

>works as expected cause probably some users want to preserve original colors of the favorite.

Fair enough. But my guess is that by far most users will expect a group's attributes to be persistent, not ephemeral. After all, they chose to set them for the "group" and not individually.

The "Group Constant Appearance" (as @vshcherb calls it) has another major advantage: **portability**. Here's the example I used:

>1. Say I have two groups, "Parks" and "Pools". The parks should be green and the pools should be blue. Logical.
>2. In `track.gpx` or `favorites.gpx` the markup is `<type>Park</type>` and `<type>Pool</type>`. Great! Now I can add a new park to the file using my desktop mapping software. I will just give it the type "Park", because my software shows Parks as green.
>3. Problem! Osmand's color information is attached _individually_ to the points as a non-standard GPX extension: `<osmand:color>#000001</osmand:color>` etc. So I'm out of luck. `<type>Park</type>` can be green in my desktop app and green in Osmand, but the POI will only be green in Osmand if it was created in Osmand. 

In other words, the desktop software stores the visual appearance of the points in its native settings, not in GPX markup. **Its GPX contains only semantic data, i.e. `<type>Park</type>`, which is completely compatible with other software.** I believe this is the best practice for managing visual style and Osmand should move to this way of doing things. 

TLDR: In Osmand, "group" equates to GPX's "type". For simplicity, and for compatibility with other software, "type" should be stylable.

**Comment by Friptick** [comment]
Thanks for the reply @Zirochkabila 

To re-clarify, I am asking for Osmand to work the same way as "desktop software" here:
>the desktop software stores the visual appearance of the points in its native settings, not in GPX markup. This way, the GPX contains only semantic data, i.e. <type>Park</type>, which is completely compatible with other software. I believe this is the best practice for managing visual style and Osmand should move to this way of doing things.

Attaching the colors and shapes and so on to individual points in the GPX markup is like doing `style="color: red;"`on HTML elements instead of in a CSS stylesheet. In GPX there is a `<type>` element and Osmand uses it. So the visual styling for `<type>Park</type>` should be in the settings somewhere.

That way, I can set `Park=blue` somewhere. And when I create an element with `<type>Park</type>` in some other software, Osmand will know that the point has to be blue.

Separating presentation and content is the only way to solve GPX portability.

**Comment by vshcherb** [comment]
I'm not sure if we are going to support such technicality. So I think to rename original request what we should support at least.

**Comment by vshcherb** [comment]
To do:
Default Group Favorite customization should be preserved:
- In case person Shares 1 Group of Favorite 
- Sends whole GPX file to a new device
- Imports from cloud to a new device

**Comment by Friptick** [comment]
As I understand it, these changes will **not** make Osmand-generated GPX more interoperable with other software.

If interoperability and portability are not a priority for Osmand (supposedly a FOSS project), it would be nice to know why.

**Comment by dmpr0** [comment]
@> If interoperability and portability are not a priority for Osmand (supposedly a FOSS project), it would be nice to know why.

Hello! If I understand you correctly you want us to answer the statement you do from OsmAnd? Could you please name these anonymous desktop software you are mention few times?

**Comment by Friptick** [comment]
@dmpr0 Right now I use GpsPrune. But I am not asking for support for any particular software package. I am advocating for interoperability. I understand you are busy but did you read my original message and the follow-up clarification?

In functional terms I believe OsmAnd users should be able to:
- create a POI ("favorite/waypoint/trackpoint") in DesktopSoftware.exe with the type "Restaurant" where all Restaurants are color `#ff0000` _and have it show up in OsmAnd as `#ff0000`_
- create a POI in OsmAnd with the same type `Restaurant` and the same color `#ff0000` _and have it show up in DesktopSoftware.exe as `#ff0000`_

This is possible. But it means moving the presentation data from OsmAnd-specific GPX markup (`<extensions><osmand:color>#ff0000</osmand:color></extensions>`) to OsmAnd's native settings.

In this way OsmAnd's GPX would be interoperable (portable). That is a good feature for FOSS software to have. It is what other software (for example GpsPrune) does: it stores the presentation data in its own settings, not in the GPX.

The title of this issue has been modified. Possibly @vshcherb did not fully understand the request, or possible he believes interoperability is impossible or undesirable. If interoperability is not a priority for OsmAnd, it would be nice to understand why.

**Comment by vshcherb** [comment]
I don't want to treat issues technical but I guess implementation will be as you said. There should be information present so default appearance for Group will be somehow preserved obviously it will be stored somewhere in extensions GPX for the whole file. 
I definitely oppose to have separate file for settings of Favorite Groups, it should be stored 1 GPX file with groups and favorites itself which will be easy to transfer and open.

**Comment by Friptick** [comment]
OK @vshcherb.

>it should be stored 1 GPX file with groups and favorites

As long as appearance is **not attached to individual POIs** then that can work, yes.

Looking in the GPX I see this already:

```
  <extensions>
    <osmand:points_groups>
      <group name="Hotel" color="#00ff00" icon="special_star" background="circle" />
      <group name="Restaurant" color="#ff0000" icon="special_star" background="circle" />
    </osmand:points_groups>
  </extensions>
```

So this seems to be what you are referring to. That can work.

In fact, if I understand correctly, only two changes are needed:
1. Apply these defaults to _all_ `<wpt><type>Hotel</type></wpt>` and `<wpt><type>Restaurant</type></wpt>` in the file, _even if they were created outside OsmAnd_
2. In `Add favorite` and `Add track waypoint`, change the UX so that when the user attributes a group via `Select group`, then the options `Select icon` and `Select color` are disabled in some way, to avoid double-styling.

**Comment by vshcherb** [comment]
Yes I think this is correct and it's already present. 
1. Probably yes it should be applied during import
2. Yes we need to change group default category UI when group is changed.

**Comment by Friptick** [comment]
>Probably yes it should be applied during import

OK but if by "applied during import" you mean "rewrite the GPX to style all the points individually", that is _not_ the right solution. I hope that is not what you mean.

It is very important to note that the `favourites.gpx` file and the `tracks` folder **can be sync targets**. This _already_ works. Users can write and delete points in those files using other software, and the changes show up in OsmAnd. **The problem is that their styling (eg., colors) is broken because of the way OsmAnd stores presentational data.** This was the whole point of my issue here.

It should _not be necessary_ to "import" or "export" anything if the presentation is stored in a interoperable way. To "import" and "export", all the user should need to do is drag the files around.

I fear there is still confusion about this whole issue. In any case, the title does not reflect what I am asking for. I want OsmAnd to be interoperable. To be fully interoperable, OsmAnd should not need to convert ("import") GPX data before displaying it.
